src/main/java/com/minecraftabnormals/abnormals_core/common/blocks/wood/WoodPostBlock.java

```diff
@@ -1,49 +1,28 @@
 package com.minecraftabnormals.abnormals_core.common.blocks.wood;
 
+import java.util.function.Supplier;
+import com.minecraftabnormals.abnormals_core.core.util.BlockUtil;
 import net.minecraft.block.Block;
 import net.minecraft.block.BlockState;
-import net.minecraft.block.IWaterLoggable;
-import net.minecraft.fluid.FluidState;
-import net.minecraft.fluid.Fluids;
-import net.minecraft.item.BlockItemUseContext;
-import net.minecraft.state.BooleanProperty;
-import net.minecraft.state.StateContainer;
-import net.minecraft.state.properties.BlockStateProperties;
+import net.minecraft.entity.player.PlayerEntity;
+import net.minecraft.item.ItemStack;
 import net.minecraft.util.math.BlockPos;
-import net.minecraft.util.math.shapes.ISelectionContext;
-import net.minecraft.util.math.shapes.VoxelShape;
-import net.minecraft.world.IBlockReader;
+import net.minecraft.world.World;
+import net.minecraftforge.common.ToolType;
 
-public class WoodPostBlock extends Block implements IWaterLoggable {
-	private static final VoxelShape SHAPE = Block.makeCuboidShape(6.0F, 0.0F, 6.0F, 10.0F, 16.0F, 10.0F);
-	private static final BooleanProperty WATERLOGGED = BlockStateProperties.WATERLOGGED;
-	
-	public WoodPostBlock(Properties properties) {
-		super(properties);
-	}
+public class WoodPostBlock extends StrippedWoodPostBlock {
+	private final Supplier<Block> block;
 
-	@Override
-	public VoxelShape getShape(BlockState state, IBlockReader worldIn, BlockPos pos, ISelectionContext context) {
-		return SHAPE;
-	}
-
-	@Override
-	public boolean propagatesSkylightDown(BlockState state, IBlockReader reader, BlockPos pos) {
-		return !state.get(WATERLOGGED);
-	}
-
-	@Override
-	public FluidState getFluidState(BlockState state) {
-		return state.get(WATERLOGGED) ? Fluids.WATER.getStillFluidState(false) : Fluids.EMPTY.getDefaultState();
+	public WoodPostBlock(Supplier<Block> suppliedBlock, Properties properties) {
+		super(properties);
+		this.block = suppliedBlock;
 	}
 	
 	@Override
-	public BlockState getStateForPlacement(BlockItemUseContext context) {
-		return this.getDefaultState().with(WATERLOGGED, context.getWorld().getFluidState(context.getPos()).getFluid() == Fluids.WATER);
+	public BlockState getToolModifiedState(BlockState state, World world, BlockPos pos, PlayerEntity player, ItemStack stack, ToolType toolType) {
+		if (toolType == ToolType.AXE)
+			return block != null ? BlockUtil.transferAllBlockStates(state, this.block.get().getDefaultState()) : null;
```

Comment:
Should `block` even possibly be null here?
All wood posts should be strippable, to my knowledge.

Reply:
This is a copy-paste of the methods that Bagel added in `WoodBlock` and `LogBlock` which I did for consistency. If wanted, however, I could remove the null check from all of them.

Reply:
I assume he did that to make the stripping optional.
In this case, the same could also be done, and no 'stripped' variant of the class is needed.